### ISSUE TITLE
Length Restrictions on Input Parameters

### ISSUE THREAD

**Issue by Aarush-Acharya** [AOSSIE-Org/Resonate#359]
### 🚀 The feature

Add length restrictions to input parameters, the limit should be such that after exceeding the limit the UI starts to overflow/fall-apart.

### Motivation, pitch

It is important to make sure we take care of every possible case as if someone wishing to break the app. All it needs is a really long Title or Description of a Room/Upcoming Room or Story/Chapter that would make the app UI overflow and break it

**Comment by M4dhav** [comment]
Hey I would like to work on this, please elaborate

**Comment by Aarush-Acharya** [comment]
Hey @M4dhav, make sure to test the UI thoroughly to identify breaking points in order to decide whats best for the project

**Comment by Aarush-Acharya** [comment]
Hey I have merged a PR please pull before working further and make sure to work on the dev branch

**Comment by Aarush-Acharya** [comment]
Hey new PR merged please pull before moving forward

**Comment by kanavsharma23** [comment]
Hey, i would also love to work on it

**Comment by kartik1112** [comment]
@Aarush-Acharya is it done? or i can work on this?

**Comment by vanshambansal** [comment]
hey would love to work on it

**Comment by M4dhav** [comment]
@vanshambansal i am working on it

**Comment by M4dhav** [comment]
Yes, I have been going through the app thoroughly and have already found and applied length restrictions to a couple of places. Some of them weren't necessary for proper display in the UI, but instead necessary for the backend as the architecture is structured to allow, for example, strings of limited length in places.

Should be done with the entire thing soon

**Comment by Aarush-Acharya** [comment]
Great @M4dhav, keep it up

**Comment by AyushJagaty** [comment]
hey I would love to work on it

**Comment by M4dhav** [comment]
@AyushJagaty I'm still working on it

**Comment by jattu8602** [comment]
hey i would love to work on it

**Comment by M4dhav** [comment]
@jattu8602 I am working on it

**Comment by Moxie-47** [comment]
Hey @M4dhav great to see that u are putting so much effort, just want to say that i've read the codebase and would like to contribute in this project in case if are not working on it.
So are u ?

**Comment by M4dhav** [comment]
@Moxie-47 Thank you for the appreciation. I am still working on this and since I have already put in a lot of work on this issue, I apologize that I cannot give this issue over to you.

**Comment by samar-703** [comment]
is this issue still open?

**Comment by M4dhav** [comment]
Yes, @samar-703 , it is open and I am working on it

**Comment by M4dhav** [comment]
@EngDann i don't think it makes sense to split this issue as Input Validation is a relatively simpler issue. Additionally I've already worked quite a bit on this issue. Thank you for your interest in contributing to Resonate. I recommend you install the app and search for additional bugs and glitches and then open PRs to fix them.

**Comment by EngDann** [comment]
@M4dhav, thanks for your response! I’ll follow your suggestion and look for other bugs and glitches to contribute.

However, I’m curious — apologies if I’m being intrusive — but if input validation is a relatively simple issue, why hasn’t it been resolved after a few months? Is there a specific reason that has hindered progress, or is there something preventing the issue from being addressed?

**Comment by M4dhav** [comment]
@EngDann there's no specific issue hindering the progress, however despite being simple enough to fix, this particular issue requires Input Validation everywhere in the app, and additionally fixing and handling overflows caused by incorrect validation which means that one has to go through the entire app comprehensively. Additionally, I've also been busy with other stuff (I'm a college student) which has caused this to take this long, but It's nearly done. I will raise a PR soon.

**Comment by EngDann** [comment]
@M4dhav I totally understand! I'm also a student, so I know how challenging it can be to juggle multiple responsibilities. Wishing you the best of luck with the fix!

**Comment by Aarush-Acharya** [comment]
Hey @M4dhav any update on this one, once done please raise the PR

**Comment by M4dhav** [comment]
@Aarush-Acharya will raise PR soon

**Comment by vikasvikku11** [comment]
is this issue already resolved ??

**Comment by M4dhav** [comment]
Hey @vnaik10 ,
Yes, this issue is resolved
